integration-cli/check_test.go

```diff
@@ -211,7 +211,7 @@ func (s *DockerSwarmSuite) AddDaemon(c *check.C, joinSwarm, manager bool) *Swarm
 		port:   defaultSwarmPort + s.portIndex,
 	}
 	d.listenAddr = fmt.Sprintf("0.0.0.0:%d", d.port)
-	err := d.StartWithBusybox("--iptables=false") // avoid networking conflicts
+	err := d.StartWithBusybox("--iptables=false", "--swarm-default-advertise-addr=lo") // avoid networking conflicts
```

Comment:
This is not the problem. It automatically adds the correct port, and there's logic in `AddDaemon` to allocate ports for the different daemons.

`TestApiSwarmForceNewCluster` only fails sometimes. I am also seeing `TestApiSwarmServicesUpdate` and `TestApiSwarmServicesMultipleAgents` fail intermittently.

I'm running into the same failures in #24943, so it does not seem to be related to the code in this PR. It looks like a problematic commit in the vendored swarmkit. I just eliminated one commit as a possibility and I'll keep trying others. It's tricky because I can't reliably reproduce any of the failures locally.